wallet/src/Pos/Wallet/Web/Methods/History.hs

```diff
@@ -204,15 +204,13 @@ addRecentPtxHistory
     :: MonadWalletWebMode m
     => CId Wal -> Map TxId TxHistoryEntry -> m (Map TxId TxHistoryEntry)
 addRecentPtxHistory wid currentHistory = do
-    pendingTxs <- getWalletPendingTxs wid
-    let candidates = toCandidates pendingTxs
-    logTxHistory "Pending" candidates
-    return $ Map.union currentHistory candidates
-  where
-    toCandidates =
-            txHistoryListToMap
-        .   mapMaybe (ptxPoolInfo . _ptxCond)
-        .   fromMaybe []
+    pendingTxs <- fromMaybe [] <$> getWalletPendingTxs wid
+    let conditions = map _ptxCond pendingTxs
+    -- show only actually pending transactions in logs
+    logTxHistory "Pending" $ mapMaybe (preview _PtxApplying) conditions
+    -- but return all transactions which are not yet in blocks
+    let candidatesList = txHistoryListToMap (mapMaybe ptxPoolInfo conditions)
```

Comment:
This is doing a double `Data.List.map` (one explicit, one as part of `txHistoryListToMap`) but I think we are saved by rewrite rules here, so that's fine 😉